### ISSUE TITLE
[ERROR] TypeError: cannot use a string pattern on a bytes-like object

### ISSUE THREAD

**Issue by loretoparisi** [nodejs/node-gyp#2121]
I'm getting this error with latest `nodehun` `nodehun@3.0.1` while building on macOS 10.14.5 (Darwin 18.6.0)

```
mbploreto:webservice loretoparisi$ node --version
v12.13.1
mbploreto:webservice loretoparisi$ npm --version
6.12.1
mbploreto:webservice loretoparisi$ node-gyp --version
v6.0.1
```

```
  CC(target) Release/obj.target/nothing/../node-addon-api/src/nothing.o
  LIBTOOL-STATIC Release/nothing.a
Traceback (most recent call last):
  File "./gyp-mac-tool", line 611, in <module>
    sys.exit(main(sys.argv[1:]))
  File "./gyp-mac-tool", line 30, in main
    exit_code = executor.Dispatch(args)
  File "./gyp-mac-tool", line 45, in Dispatch
    return getattr(self, method)(*args[1:])
  File "./gyp-mac-tool", line 248, in ExecFilterLibtool
    if not libtool_re.match(line) and not libtool_re5.match(line):
TypeError: cannot use a string pattern on a bytes-like object
make: *** [Release/nothing.a] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.6.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/loretoparisi/Documents/Projects/node_modules/nodehun
gyp ERR! node -v v12.13.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
npm WARN musixmatch-ai-console@6.2.6-20200511 No repository field.
npm WARN musixmatch-ai-console@6.2.6-20200511 No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nodehun@3.0.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nodehun@3.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
```

It seems to me that this is related to node-gyp rather than [this](https://github.com/Wulf/nodehun) module itself.

**Comment by loretoparisi** [comment]
**[UPDATE]**
According to [this](https://stackoverflow.com/questions/52987586/if-not-libtool-re-matchline-and-not-libtool-re5-matchline-typeerror-cannot) SF question this issue maybe related when using Python 3.x instead of Python 2.7 when on macOS
If this is the case, how to make node-gyp to use the old version of python. When on macOS we now have python 3.7.7 with brew by defaults.

**Comment by cclauss** [comment]
Please upgrade `gyp ERR! node-gyp -v v5.0.5`

**Comment by loretoparisi** [comment]
@cclauss thanks, I can only see
```
ip-192-168-1-168:Projects loretoparisi$ which node-gyp
/usr/local/bin/node-gyp
ip-192-168-1-168:Projects loretoparisi$ node-gyp --version
v6.0.1
```

**Comment by cclauss** [comment]
Please look through your log above:
```
gyp ERR! node -v v12.13.1
gyp ERR! node-gyp -v v5.0.5
```

**Comment by loretoparisi** [comment]
@cclauss thank you. So if I do a `locate node-gyp` I cannot find the offender version `v5.0.5` under path like

```
/usr/local/n/versions/node/
/usr/local/lib/node_modules/npm/node_modules/node-gyp
/usr/local/Cellar/node/
/Users/loretoparisi/.node-gyp/
```

etc. so what is forcing this command `node-gyp` during the build?

**Comment by cclauss** [comment]
/Users/loretoparisi/Documents/Projects/node_modules/node-gyp/gyp ?

**Comment by loretoparisi** [comment]
while I have all these (not sure why actually!)

```
/Users/loretoparisi/.node-gyp/
10.0.0/  10.1.0/  10.15.3/ 11.1.0/  6.9.4/   8.11.1/  8.11.2/  8.12.0/  8.9.1/   8.9.4/   9.11.1/  
```

I think I have found it here
```
 /usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp --version
ERROR:root:code for hash md5 was not found.
Traceback (most recent call last):
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/hashlib.py", line 147, in <module>
    globals()[__func_name] = __get_hash(__func_name)
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/hashlib.py", line 97, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type md5
ERROR:root:code for hash sha1 was not found.
Traceback (most recent call last):
```

**Comment by Wulf** [comment]
hey @loretoparisi :)

I ran into this issue myself -- turns out I just needed to updated NPM itself (which has `node-gyp` as a dependency)

```
npm i -g npm
```

**Comment by tekumara** [comment]
Thanks @wulf, `npm i -g npm` fixed this for me! (I'm using node v11.15.0 on Darwin 19.6.0)

**Comment by loretoparisi** [comment]
@Wulf great, thank you! in fact it worked with `node v12.16.1`, `npm v6.13.4`, `node-gyp v6.0.1`. Closing then.
